src/nucleus/tt.ml

```diff
@@ -115,215 +115,220 @@ let mention_atoms a e =
 let mention a e =
   { e with assumptions = Assumption.union e.assumptions a }
 
-(** Manipulation of variables *)
 
-let rec instantiate_ty_abstraction :
-  'a. (term list -> ?lvl:int -> 'a -> 'a) ->
-  term list -> ?lvl:int -> 'a ty_abstraction -> 'a ty_abstraction
-  = fun instantiate_v es ?(lvl=0) ((x,u), v) ->
-  let u = instantiate_ty es ~lvl u in
-  let lvl = lvl+1 in
-  let v = instantiate_v es ~lvl v in
-  ((x,u),v)
+let rec gather_assumptions_ty_abs : 'a. ('a -> Assumption.t) -> 'a ty_abstraction -> Assumption.t = fun gather_v ((x,u),v) ->
+  let a1 = gather_assumptions_ty u in
+  let a2 = gather_v v in
+  Assumption.union a1 (Assumption.bind 1 a2)
 
-and instantiate es ?(lvl=0) ({term=e';assumptions;loc;} as e) =
-  if es = [] then e else
-  let assumptions =
-    Assumption.instantiate (List.map (fun e -> e.assumptions) es) lvl assumptions
```

Comment:
Here we need to use gather_term instead of only getting the assumptions on the top constructor.